flexget/plugins/input/sonarr_emit.py

```diff
@@ -0,0 +1,132 @@
+    get the relevant data from Sonarr.
+
+    Syntax:
+
+    sonarr_emit:
+      base_url=<value> (Required)
+      port=<value> (Default is 80)
+      api_key=<value> (Required)
+      include_ended=<yes|no> (Default is yes)
+      only_monitored=<yes|no> (Default is yes)
+      page_size=<value> (Default is 50)
+
+    Page size determines the amount of results per each API call.
+    Higher value means a bigger response. Lower value means more calls.
+    Should be changed if there are performance issues.
+
+
+    Usage: (Example with discover)
+
+    discover_from_sonarr_task:
+      discover:
+        what:
+          - sonarr_emit:
+              base_url: '{{ secrets.credentials.sonarr.url }}'
+              port: 8989
+              api_key: '{{ secrets.credentials.sonarr.api_key }}'
+              include_ended: false
+        from:
+          - kat:
+              verified: yes
+      all_series: yes
+      download: c:\bla\
+
+    Usage: (Example with set_series_begin)
+
+    set-series-begin-from-sonarr:
+      sonarr_emit:
+        base_url: '{{ secrets.credentials.sonarr.url }}'
+        port: 8989
+        api_key: '{{ secrets.credentials.sonarr.api_key }}'
+        include_ended: false
+      accept_all: yes
+      set_series_begin: yes
+    """
+    schema = {
+        'type': 'object',
+        'properties': {
+            'base_url': {'type': 'string'},
+            'port': {'type': 'number', 'default': 80},
+            'api_key': {'type': 'string'},
+            'include_ended': {'type': 'boolean', 'default': True},
+            'only_monitored': {'type': 'boolean', 'default': True},
+            'page_size': {'type': 'number', 'default': 50}
+        },
+        'required': ['api_key', 'base_url'],
+        'additionalProperties': False
+    }
+
+    # Function that gets a page number and page size and returns the responding result json
+    def get_page(self, task, config, page_number):
+        parsedurl = urlparse(config.get('base_url'))
+        url = '%s://%s:%s%s/api/wanted/missing?page=%d&pageSize=%d&sortKey=series.title&sortdir=asc' \
+              % (parsedurl.scheme, parsedurl.netloc, config.get('port'),
+                 parsedurl.path, page_number, config.get('page_size'))
+        headers = {'X-Api-Key': config['api_key']}
+        try:
+            json = task.requests.get(url, headers=headers).json()
+        except RequestException as e:
+            raise plugin.PluginError('Unable to connect to Sonarr at %s://%s:%s%s. Error: %s'
+                                     % (parsedurl.scheme, parsedurl.netloc, config.get('port'),
+                                        parsedurl.path, e))
+        return json
+
+    def on_task_input(self, task, config):
+        json = self.get_page(task, config, 1)
+        entries = []
+        pages = int(json['totalRecords'] / config.get('page_size')) + (json['totalRecords'] % config.get('page_size') > 0)  # Sets number of requested pages
```

Comment:
I think there was a reason I did it like that, but I can't figure out why. I'll change this.